### ISSUE TITLE
User Story -- Owner/Shopper Contributor

### ISSUE THREAD

**Issue by hershyd** [Compuskills-Jerusalem/shuk-routing#3]
As a store owner in the shuk , 
I can type my store name and contribute my products & information about it, To the app,
So that my products & prices would be up to date

**Comment by ttafet** [comment]
Not only as a store owner. A contributor can also be a consumer.

**Comment by ykopstein** [comment]
@ttafet What if the store owner doesn't want random consumers to be allowed to update his information?  His prices could be sabotaged by competitors, misentered, etc.  With @hershyd option, the store owner makes the decisions.

**Comment by ttafet** [comment]
agree, but the Waze model uses data from all travelers... I agree, this
isn't the same use case, but we want to democratize the price reporting...
sorry for the rush, I need to catch a cab... let's do it Hershy's way for
the 1st release and see what we've got.

**Comment by ShmielB** [comment]
1) 
@hershyd you say As a store owner.... I **can type** my store name . 
I think it will be better if we will have a drop down (when you click or hover over stall name) that will display all stalls that are registered  , also like this if we do as @ttafet said it will be much easier for our consumers to see all the stall names .

2)
I think like @ttafet said like the Waze model , and for the concern of @ykosbie we can add a log in so this way every update will note who added it and other users can spam him if they see he added false info .

**Comment by ykahan** [comment]
@hershyd  @ttafet  @ykosbie 
Perhaps it could be set up so that when the owner enters info, it will be flagged as owner-supplied info?  This way, he can make sure that his stall appears at all (instead of relying on random consumers to register it), but people who see the info will know that it was posted by the owner, who maybe has a vested interested in getting you to think that it's better produce than it really is.

**Comment by hershyd** [comment]
As a store owner/shopper in the shuk ,
I can select my/a store name from the drop down menu, and contribute products & information about it, To the app,
So that products & prices would be up to date.
----
and it will be flagged as owner-supplied /consumer-supplied info,
That means every owner/consumer who would like to contribute information will need to be logged in and verified, and every update will note who added it and other users can spam him if they see he added false info, like @ShmielB suggested.

**Comment by ShmielB** [comment]
@hershyd  i think we should define this user story into **Requirements** and **Architecture**  (classes) so we can divide the tasks and start implementing it .

**Comment by hershyd** [comment]
**Requirements**

1) Login box
2) Add auth for the login  
3) Show drop down menu to get all the stores available
4) Handle the drop down menu being clicked
5) Add a database connection
6) Make a query to get all the stores available for the drop down menu
7) Send back to the client all the stores
8) Display the stores on screen
9) Handle the click on a specific store and display it on screen
10) Display input boxes to add the info like prices, quality, etc
11) submit button to send the data to the server 
12) Make a query to send all the info added, into the database
13) Update the database

**Comment by ShmielB** [comment]
**About finalizing this idea of the login** in my comment before i suggested that it should note on each contribute or update who made it and there should be a way to spam that user so we could abandon that user from contributing    in such a way we dont have to worry about @ykosbie problem and and @Jatudrei  problem  and in the same time implement  what @ttafet said , unless @Jatudrei  you think we should have a separate flag for store owners and for consumers . 

### I spoke it through with @hershyd the log in processes should be broken up to its own requirements .

### Login / Sign up Requirements, covering reqe. 1 , 2 from above .
_the way we go now that each user will be flagged_  

1) Show login page ( popup ) when clicking to add or update info if not logged-in yet

on the page:

 _Login if already have an account_

        1) Show login ( Username / Password ) input box , Go button
        2) Handel go button 
        3) Connect to db 
        4) Send request to db 
        5) Return db result
        6) Show the previous page (the page they were )

  _Sign up ( Create new account )_

      1)  Create account button ( under the login input box )
      2) Handel button
      3) Show create new account page 
            1) Show input boxes for user info
            2) Show button to submit 
            3) Connect to db
            4) Handel the submit button
            5) Send Http request to db
            6) Show privies page ( return to the page they wanted to go )

2) Show logged in as with user name ( on place of the sign-in button , top right ) 
3) Connect each add or edit to the user ( needs to be divided by its self )
       

n.b
@ykahan   we need to add user table to the db .

**Comment by hershyd** [comment]
**classes & methods**
1) CommodityStall controller 
   - Create (action method GET)
   - Create (action method POST)

2) CommodityStall Repository
   - CreateCommStall
   - CommodityStallSave

3) Stall Repository
   - GetStallNames

**Comment by ttafet** [comment]
@hershyd  - could you please flesh out "controller" a bit more: e.g. controller of what?

**Comment by hershyd** [comment]
Requirement # 1, 2,  I am not covering here, i am going to do a extra user story for auth, and full requirements for it.

Requirement # 3, 10, 11,  Is done in the create view from the CommodityStall controller.
Requirement # 4, 8, 9,  Is done with the html helper DropDownListFor in the create view from the 
         CommodityStall controller.

Requirement # 6, 7,  Is done in class StallRepository method GetStallNames().
Requirement # 12, 13, 5  Is done in the class CommodityStallRepository method CommodityStallSave().

**Comment by ShmielB** [comment]
@hershyd  Lookes grate , and about opening a new issue for the login is also a good idea like that we can discus everything about it there , i added some req. for the log in above but ill copy it to the new Login Issue . 

And i think we should also add an issue for the other part what we discussed in this issue **about flagging each add to the user** , the idea was discussed but no req. or implementation was done .
